ui/src/components/song/SongEditorDesktopView.tsx

```diff
@@ -0,0 +1,304 @@
+import { FC, ReactElement, useState } from 'react';
+import {
+  Container,
+  Box,
+  Typography,
+  TextField,
+  FormControl,
+  Stack,
+  TextareaAutosize,
+  Autocomplete,
+  Button,
+  Divider,
+  Toolbar,
+  Chip,
+  Alert,
+  AlertTitle,
+  Snackbar,
+  Fade,
+  Grid,
+} from '@mui/material';
+import { musicKeysOptions, tempoOptions } from '../../constants';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { SongEditorFields, SongEditorProps } from '#/types/song.types';
+import SongHelpDialog from './SongHelpDialog';
+import AutocompleteInput from '../custom/AutocompleteInput';
+import axios from 'axios';
+
+// ICONS
+import EditIcon from '@mui/icons-material/Edit';
+import AddIcon from '@mui/icons-material/Add';
+import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
+import InfoIcon from '@mui/icons-material/Info';
+import LibraryMusicIcon from '@mui/icons-material/LibraryMusic';
+
+const SongEditorDesktopView: FC<SongEditorProps> = ({ actionOnEditor }) => {
+  // STATES
+  const [themes, setThemes] = useState<string | string[] | null>([]);
+  const [tempo, setTempo] = useState<string | string[] | null>([]);
+  const [recommendedKeys, setRecommendedKeys] = useState<string | string[] | null>([]);
+
+  const [successSnackbarOpen, setSuccessSnackbarOpen] = useState<boolean>(false);
+  const [invalidSong, setInvalidSong] = useState<string>('');
+
+  // FORM HANDLER
+  const { register, handleSubmit, formState } = useForm<SongEditorFields>();
+  const { errors } = formState;
+
+  // TODO: get theme options from database
+  const themeOptions: string[] = ['Love', 'Faith', 'Hope', 'Joy', 'Peace', 'Grace'];
+
+  // editor mode is either ADD NEW or EDIT. default is ADD NEW
+  const editorMode = (actionOnEditor: string): ReactElement => {
+    return (
+      <Typography
+        variant="h3"
+        fontWeight="700"
+        color="primary"
+        sx={{ display: 'flex', alignItems: 'center' }}
+        gap={1}
+        mx={2}
+      >
+        {actionOnEditor === 'EDIT' ? <EditIcon color="primary" /> : <AddIcon color="primary" />}
+        {actionOnEditor ? actionOnEditor.toUpperCase() : 'ADD NEW'} SONG
+      </Typography>
+    );
+  };
+
+  const handleSaveSong: SubmitHandler<SongEditorFields> = async (data) => {
+    try {
+      const payload = await axios.post('/api/songs/create', {
+        artist: data.artist,
+        title: data.title,
+        themes: themes,
+        tempo: tempo,
+        year: data.year,
+        code: data.code,
+        originalKey: data.originalKey,
+        recommendedKeys: recommendedKeys,
+        chordLyrics: data.chordLyrics,
+        // TODO: figure out how to retrieve proper lyrics preview
```

Comment:
just an idea for future cycles, since we are gonna force the user to make the chord lyrics with #verse1, #chorus, etc. Maybe can make the preview from the start of the first # to right before the second #?